src/Runner.Worker/ActionCommandManager.cs

```diff
@@ -183,6 +183,41 @@ public sealed class SetEnvCommandExtension : RunnerService, IActionCommandExtens
 
         public void ProcessCommand(IExecutionContext context, string line, ActionCommand command, ContainerInfo container)
         {
+
+            var serverUrl = context.GetGitHubContext("server_url");
+            var isGHES = false;
+            if(!String.IsNullOrWhiteSpace(serverUrl) && serverUrl != "https://github.com")
+            {
+                isGHES = true;
+            }
+            
+            var allowUnsecureCommands = false;
+            if (!String.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable(Constants.Variables.Actions.AllowUnsupportedCommands)) 
+                && Environment.GetEnvironmentVariable(Constants.Variables.Actions.AllowUnsupportedCommands).ToUpper() == "TRUE")
```

Comment:
The usual pattern we use in the runner is to use the `TryParse()` family of methods to convert from the string to the object type of choice. For example: https://github.com/actions/runner/blob/2709cbc0eae592e6004b016c52c04382f19c7683/src/Runner.Sdk/Util/IOUtil.cs#L398